agent/recognition/binarymatch.py

```
@@ -568,6 +541,48 @@ def _try_recognition(self, context: Context, node_name: str,
+# vert(纵横比)是模糊态区分"竖条 vs 圆形高光"的主轴，故权重最高；white 仅作辅证。
+_W_GAP = 0.45      # 连续断层：清晰帧(PC)主力，带两侧门，模糊归零
+_W_VERT = 0.35     # 纵横比：模糊地板主力，扛等高实心柱、排圆形高光
+_W_WHITE = 0.15    # 偏白对比(S 为主、V 为辅，相对红环)：抗压暗的辅证
+_W_CENT = 0.05     # 居中：小补充
+_DEFAULT_MIN_CONF = 0.25   # 精准 ROI 感叹号默认阈值；大 ROI 泛找请显式调高
+
+# 同一检测点(节点名+ROI)两次落盘的最小间隔(秒)，防 next 自循环刷屏；RDD_DUMP_INTERVAL 可调
+try:
+    _DUMP_MIN_INTERVAL = float(os.environ.get("RDD_DUMP_INTERVAL", "2.0"))
+except (TypeError, ValueError):
+    _DUMP_MIN_INTERVAL = 2.0   # 环境变量非法时兜底，避免 import 阶段崩溃中断 Agent 启动
```

Ruff (0.15.15)

[warning] Comment contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF003)